bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/type/ZwaveJSTypeGeneratorImpl.java

```diff
@@ -199,53 +199,59 @@ private ConfigDescriptionParameter createConfigDescription(ConfigMetadata detail
+            logger.trace("Node {}. Ignoring channel with Id: {} (ignored command class)", details.nodeId, details.id);
             return result.channels;
         }
-        logger.trace("Node {} building channel with Id: {}", details.nodeId, details.id);
+        logger.trace("Node {}. building channel with Id: {}", details.nodeId, details.id);
         logger.trace(" >> {}", details);
 
         ChannelUID channelUID = new ChannelUID(thingUID, details.id);
-        Configuration newChannelConfiguration = buildChannelConfiguration(details);
+        Configuration channelConfiguration = buildChannelConfiguration(details);
 
-        // Check for existing channel
+        // Try to reuse or update an existing channel
         Channel existingChannel = result.channels.get(channelUID.getId());
+        ChannelTypeUID channelTypeUID = generateChannelTypeUID(details);
+        ChannelType channelType = getOrGenerate(channelTypeUID, details);
+
+        String label = details.label;
         if (existingChannel != null) {
+            // Update configuration and label if needed
             Configuration existingConfig = existingChannel.getConfiguration();
-            if (ChannelMetadata.isSameReadWriteChannel(existingConfig, newChannelConfiguration)) {
-                updateReadWriteProperties(existingConfig, details);
+            updateReadWriteProperties(existingConfig, details);
 
-                ChannelTypeUID newChannelTypeUID = generateChannelTypeUID(details);
-                ChannelBuilder builder = ChannelBuilder.create(existingChannel).withConfiguration(existingConfig);
-
-                if (!newChannelTypeUID.equals(existingChannel.getChannelTypeUID())) {
-                    ChannelType newChannelType = getOrGenerate(newChannelTypeUID, details);
-                    if (newChannelType != null) {
-                        builder.withAcceptedItemType(newChannelType.getItemType()).withType(newChannelType.getUID());
-                    }
-                }
+            if (details.writable) {
+                existingConfig.put(BindingConstants.CONFIG_CHANNEL_ITEM_TYPE, details.itemType);
+                label = existingChannel.getLabel() != null ? existingChannel.getLabel() : label;
+            }
+            // If the channel type UID has changed and the channel is not writable, keep the old type UID
+            if (!channelTypeUID.equals(existingChannel.getChannelTypeUID()) && !details.writable) {
+                channelTypeUID = existingChannel.getChannelTypeUID();
+            }
+            channelConfiguration = existingConfig;
+            logger.debug("Node {}. Channel {}: existing channel updated", details.nodeId, details.id);
+        }
 
-                result.channels.put(details.id, builder.build());
+        // if necessary add or update the entry in our ZwaveJSTypeGeneratorResult's map of ColorCapabilities
+        updateColorCapabilities(thingUID, details, result);
```

Comment:
Maybe you are calling `updateColorCapabilities()` too soon? It seems to me that you are creating the new channel via `builder.build()` in line 256 below, so `updateColorCapabilities()` would not include that channel..